internal/broker/schemas.go

```diff
@@ -0,0 +1,347 @@
+package broker
+
+import (
+	"io"
+
+	pkg "github.com/kyma-project/kyma-environment-broker/common/runtime"
+	"github.com/kyma-project/kyma-environment-broker/internal/provider/configuration"
+	"github.com/pivotal-cf/brokerapi/v12/domain"
+)
+
+type SchemaService struct {
+	planSpec          *configuration.PlanSpecifications
+	providerSpec      *configuration.ProviderSpec
+	defaultOIDCConfig *pkg.OIDCConfigDTO
+
+	ingressFilteringFeatureFlag bool
+	ingressFilteringPlans       EnablePlans
+
+	cfg Config
+}
+
+func NewSchemaService(providerConfig io.Reader, planConfig io.Reader, defaultOIDCConfig *pkg.OIDCConfigDTO, cfg Config, ingressFilteringEnabled bool, ingressFilteringPlans EnablePlans) (*SchemaService, error) {
+	planSpec, err := configuration.NewPlanSpecifications(planConfig)
+	if err != nil {
+		return nil, err
+	}
+	providerSpec, err := configuration.NewProviderSpec(providerConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return &SchemaService{
+		planSpec:                    planSpec,
+		providerSpec:                providerSpec,
+		defaultOIDCConfig:           defaultOIDCConfig,
+		cfg:                         cfg,
+		ingressFilteringFeatureFlag: ingressFilteringEnabled,
+		ingressFilteringPlans:       ingressFilteringPlans,
+	}, nil
+}
+
+func (s *SchemaService) Plans(plans PlansConfig, platformRegion string, cp pkg.CloudProvider) map[string]domain.ServicePlan {
+	awsCreateSchema := s.AWSSchema(platformRegion, false)
```

Comment:
This repetitive pattern is error prone. It is easy to use the wrong boolean flag in the invocation.
I do not have perfect solution but using single function per plan returning two schemas (create and update) and encapsulating calls with boolean `update` appeals to me. 
But this is very subjective.